zyte_common_items/pages.py

```diff
@@ -54,6 +61,33 @@ def _get_metadata_class(cls: type) -> Optional[Type[MetadataT]]:
     return get_generic_param(cls, HasMetadata)
 
 
+class PriceMixin(FieldsMixin):
+    """Provides price-related field implementations."""
+
+    _parsed_price: Optional[Price] = None
+
+    async def _get_parsed_price(self):
```

Comment:
Would it work if we use cached_method decorator, instead of using an explicit _parsed_price attribute? Or are there issues with slots, etc. which made you go with the custom route?

Reply:
`price_processor` needs to assign the value directly, so `cached_method` didn't look like it will be enough here.